migrations/migrate63

```diff
@@ -0,0 +1,181 @@
+-- Create the needed tables.
+CREATE TABLE "project_notification" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "created_ts" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "type" VARCHAR NOT NULL,
+    "to" INT8 NOT NULL,
+    "project" INT8 NOT NULL,
+    "content" VARCHAR NOT NULL,
+    "archived" BOOLEAN NOT NULL
+);
+
+CREATE TABLE "project_notification_email" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "created_ts" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "type" VARCHAR NOT NULL,
+    "to" INT8 NOT NULL,
+    "project" INT8 NOT NULL,
+    "content" VARCHAR NOT NULL
+);
+
+CREATE TABLE "project_notification_pref" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "user" INT8 NOT NULL,
+    "project" INT8 NOT NULL,
+    "type" VARCHAR NOT NULL,
+    "delivery" VARCHAR NOT NULL
+);
+
+CREATE TABLE "user_notification" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "created_ts" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "type" VARCHAR NOT NULL,
+    "to" INT8 NOT NULL,
+    "content" VARCHAR NOT NULL,
+    "archived" BOOLEAN NOT NULL
+);
+
+CREATE TABLE "user_notification_email" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "created_ts" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "type" VARCHAR NOT NULL,
+    "to" INT8 NOT NULL,
+    "content" VARCHAR NOT NULL
+);
+
+CREATE TABLE "event_user_notification_sent" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "ts" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "notification" INT8 NOT NULL
+);
+
+CREATE TABLE "event_project_notification_sent" (
+    "id" SERIAL PRIMARY KEY UNIQUE NOT NULL,
+    "ts" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "notification" INT8 NOT NULL
+);
+
+
+-- Change preferences having both 'NotifDeliverWebsite' and
+-- 'NotifDeliverEmail' to 'NotifDeliverWebsiteAndEmail'.
+UPDATE "user_notification_pref" unp
+SET "delivery" = 'NotifDeliverWebsiteAndEmail'
+FROM (
+    SELECT "user", "type", "project"
+    FROM "user_notification_pref"
+    GROUP BY "user", "type", "project"
+    HAVING bool_or("delivery" = 'NotifDeliverEmail')
+       AND bool_or("delivery" = 'NotifDeliverWebsite')
+) AS subq
+WHERE subq.user = unp.user
+  AND subq.type = unp.type
+  AND subq.project IS NOT DISTINCT FROM unp.project
+;
+
+-- Remove the resulting duplicates
+-- (see https://wiki.postgresql.org/wiki/Deleting_duplicates).
+DELETE FROM "user_notification_pref"
```

Comment:
If you had use "SELECT DISTINCT" in the subquery above, could you have skipped this step?